### PR TITLE
WT-7452 Add addtional verbose messaging to rollback to stable.

### PR DESCRIPTION
**Pull request by luke-pearson**

### CODE REVIEW

**Comment by sulabhM** [comment]
I like this version better, but would have preferred `__wt_verbose_progress` as a function and not as a macro. Any specific reason why you decided to have a macro instead of a function ?

**Comment by luke-pearson** [comment]
> 
> 
> I like this version better, but would have preferred `__wt_verbose_progress` as a function and not as a macro. Any specific reason why you decided to have a macro instead of a function ?

From the comment of `__wt_verbose` we can't inline the function and take varargs, which would add a performance overhead when checking that the flag is set. Additionally I think the flag `#flag` print doesn't work the same. I did try that and this is what I got, which I thought was uglier:

```
/*
 * __wt_verbose_progress--
 *     Verbose message.
 */
static void
__wt_verbose_progress(WT_SESSION_IMPL *session, struct timespec *start_time, uint64_t *msg_count, bool closing, uint64_t verbose_type, const char *fmt, ...)
  WT_GCC_FUNC_ATTRIBUTE((format(printf, 2, 3))) WT_GCC_FUNC_ATTRIBUTE((cold))
{
    va_list ap;
    struct timespec cur_time;
    uint64_t time_diff;

    __wt_epoch(session, &cur_time);

    /* Time since the rollback started */
    time_diff = WT_TIMEDIFF_SEC(cur_time, *start_time);

    if (closing || (time_diff / WT_PROGRESS_MSG_PERIOD) > *msg_count) {
        va_start(ap, fmt);
        __wt_verbose(session, verbose_type, fmt, ap);
        va_end(ap);

        ++(*msg_count);
    }

```

Comment for reference: 
https://github.com/wiredtiger/wiredtiger/blob/2834b46119c9cccdc55dc82ff79d6ac4f3fa97e7/src/include/error.h#L175

**Comment by luke-pearson** [comment]
There was another solution implemented for this see: #6573

File: src/btree/bt_sync.c [hunk 1]
```diff
@@ -650,7 +650,11 @@ __wt_sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
 
                 /* Periodically log checkpoint progress. */
                 if (conn->ckpt_write_pages % 5000 == 0)
-                    __wt_checkpoint_progress(session, false);
+                    __wt_verbose_progress(session, WT_VERB_CHECKPOINT_PROGRESS,
+                      conn->ckpt_timer_start, &conn->ckpt_progress_msg_count, true,
+                      "Checkpoint has been running for %" PRIu64 " seconds and wrote: %" PRIu64
+                      " pages (%" PRIu64 " MB)",
+                      time_diff, conn->ckpt_write_pages, conn->ckpt_write_bytes / WT_MEGABYTE);
```
Comment (sulabhM) [inline src/btree/bt_sync.c:657]: Ah! did not notice before, referring to `time_diff` here where it is not defined. It is actually defined and used in the macro itself. That doesn't look clean. If we are going to use a common macro, maybe the message provided through the invocation of the macro can be appended to a generic message inside the macro. The message inside the macro could talk about the operation that has been running for x duration. I am not sure how hard that would be without changing the existing message we print for checkpoint duration. I am also not sure if it is okay to change the words for an existing message.

**Review by sulabhM — CHANGES_REQUESTED** [review]

File: src/btree/bt_sync.c (hunk 1 above)
Comment (sulabhM) [reply src/btree/bt_sync.c:657]: I thought more about if there is a clean way of doing this using a macro, and without changing the message we already have for the checkpoint, it seems overly complicated. We should evaluate if we can change the wording of the checkpoint message, if not maybe switch to the other branch you created?

Comment (luke-pearson) [reply src/btree/bt_sync.c:657]: Well the messages provided were intended to be use-case specific I don't think we can generalize them between the two if that was the idea?

**Comment by luke-pearson** [comment]
Closing as this doesn't end up being cleaner than the other solution.

**Pull request closed without merging**